// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

// Diskio plug-in configuration, passed to plug-in by plug-in service.
message DiskioConfig {
    int32 unspeci_fied = 1; // Reserved field, not used
    enum IoReportType {
        UNSPECIFIED = 0;
        IO_REPORT = 1;
        IO_REPORT_EX = 2;
    }
    IoReportType report_io_stats = 2;
}
